portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
+/*
+ * Returns a random fun fact about me until all facts are exhausted.
+ * 
+ * REQUIRES: none
+ * ENSURES: The "facts" container is replaced with a randomly chosen fun fact as
+ *          defined in the facts array, and that fact is removed from facts. If
+ *          the facts array is empty, the "generate-fact" container is removed.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
+var getRandomFact = (function() {
```

Comment:
This is OK, but I prefer using "let" over "var" in JS since it obeys its scope. "var," on the other hand, can be used within or outside of its defined scope, which can occasionally lead to some difficult bugs.

Reply:
Ok, I'm still getting the hang of the different use cases between let and var (and const, too), so I can make that change.